wrangler-core/src/main/java/co/cask/directives/parser/ParseTimestamp.java

```diff
@@ -0,0 +1,137 @@
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A Executor to parse timestamp as date.
+ */
+@Plugin(type = Directive.Type)
+@Name("parse-timestamp")
+@Categories(categories = {"parser", "date"})
+@Description("Parses column values representing unix timestamp as date.")
+public class ParseTimestamp implements Directive {
+  public static final String NAME = "parse-timestamp";
+  private static final Set<TimeUnit> SUPPORTED_TIME_UNITS = EnumSet.of(TimeUnit.SECONDS, TimeUnit.MILLISECONDS,
+                                                                       TimeUnit.MICROSECONDS);
+  private String column;
+  private TimeUnit timeUnit;
+
+  @Override
+  public UsageDefinition define() {
+    UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
+    builder.define("column", TokenType.COLUMN_NAME);
+    builder.define("timeunit", TokenType.TEXT, Optional.TRUE);
+    return builder.build();
+  }
+
+  @Override
+  public void initialize(Arguments args) throws DirectiveParseException {
+    this.column = ((ColumnName) args.value("column")).value();
+    this.timeUnit = TimeUnit.MILLISECONDS;
+
+    if (args.contains("timeunit")) {
+      String unitValue = ((Text) args.value("timeunit")).value();
+      this.timeUnit = getTimeUnit(unitValue.toUpperCase());
+    }
+  }
+
+  @Override
+  public void destroy() {
+    // no-op
+  }
+
+  @Override
+  public List<Row> execute(List<Row> rows, ExecutorContext context)
+    throws DirectiveExecutionException, ErrorRowException {
+    for (Row row : rows) {
+      int idx = row.find(column);
+      if (idx != -1) {
+        Object object = row.getValue(idx);
+        // If the data in the cell is null or is already of date format, then
+        // continue to next row.
+        if (object == null || object instanceof ZonedDateTime) {
+          continue;
+        }
+
+        long longValue = getLongValue(object);
+        ZonedDateTime zonedDateTime = getZonedDateTime(longValue, timeUnit, ZoneId.ofOffset("UTC", ZoneOffset.UTC));
+        row.setValue(idx, zonedDateTime);
+      }
+    }
+    return rows;
+  }
+
+  private static TimeUnit getTimeUnit(String unitValue) throws DirectiveParseException {
+    if (EnumUtils.isValidEnum(TimeUnit.class, unitValue) &&
+      SUPPORTED_TIME_UNITS.contains(TimeUnit.valueOf(unitValue))) {
+      return TimeUnit.valueOf(unitValue);
+    }
+
+    throw new DirectiveParseException(String.format("Provided time unit %s is not supported. Supported time " +
+                                                      "units are %s", unitValue, SUPPORTED_TIME_UNITS));
+  }
+
+  private long getLongValue(Object object) throws ErrorRowException {
+    long longValue;
+    if (object instanceof Long) {
+      longValue = (long) object;
+    } else if (object instanceof String) {
+      longValue = Long.parseLong((String) object);
```

Comment:
if this throws an exception, should we catch it and throw an ErrorRowException?

Is the contract that ErrorRowExceptions are emitted as error records?

Reply:
yes `ErrorRowException` is thrown to emit records to error collector.